src/util.ts

```diff
@@ -130,3 +130,48 @@ export function joinAll(a: ArrayBuffer[]): ArrayBuffer {
 
     return buffer;
 }
+
+// Consumes data:
+//   PrivateToken challenge="abc...", token-key="123...",
+//   PrivateToken challenge="def...", token-key="234...",
+//   PrivateToken challenge=ghi..., token-key=345...
+// Parse WWW-Authenticate according to RFC9110 Section 11.6.1 https://www.rfc-editor.org/rfc/rfc9110#section-11.6.1
+export function parseWWWAuthenticate(header: string): string[] {
+    const ALPHA = 'A-Za-z';
+    const DIGIT = '0-9';
+    const tokenChar = `!#$%&'*+\\-\\.^_\`|~${DIGIT}${ALPHA}`;
+    const tchar = '[' + tokenChar + ']';
+    const OWS = '[ \\t]*';
+    const RWS = '[ \\t]+';
+    const BWS = OWS;
+    const qdtext = '[\\t\\x21\\x23-\\x5B\\x5D-\\x7E\\x80-\\xFF]';
+    const quotedPair = '\\\\([ \\t' + tokenChar + '\\x21-\\x7E\\x80-\\xFF])';
+    const quotedString = `"(?:${qdtext}|${quotedPair})*"`;
+    const token = tchar + '+';
+    const authParam = `${token}${BWS}=${BWS}(?:${token}|${quotedString})`;
+    const authScheme = token;
+    const token68 = `[\\/+\\-\\._~${DIGIT}${ALPHA}]+=*`;
+    const challenge = `${authScheme}(?:${RWS}(?:${token68}|${authParam}(?:${OWS},${OWS}${token68}|${authParam})*))?`;
```

Comment:
Spec says
```
challenge   = auth-scheme [ 1*SP ( token68 / #auth-param ) ]
```
which basically [expands](https://www.rfc-editor.org/rfc/rfc9110#abnf.extension) to:
```
challenge   = auth-scheme [ 1*SP ( token68 / (auth-param *( OWS "," OWS auth-param ) ) ) ]
```
so the `${RWS}` should be space to not match tabs, but one of more spaces. That is also assumed by the `PrivateToken ` check above. Given that more than one space can be matched, the code above should probably be updated.
```suggestion
    const challenge = `${authScheme}(?: +(?:${token68}|${authParam}(?:${OWS},${OWS}${token68}|${authParam})*))?`;
```

Reply:
updated